### PR TITLE
KAA-819 Smart persistence for C++ SDK

### PR DESCRIPTION
**Pull request by forGGe**
Minor refactoring applied.

### CODE REVIEW

File: client/client-multi/client-cpp/impl/ClientStatus.cpp [hunk 1]
```diff
@@ -566,6 +546,9 @@ void ClientStatus::read()
 
 void ClientStatus::save()
 {
+    if (!hasUpdate_)
```
Comment (ghost) [inline client/client-multi/client-cpp/impl/ClientStatus.cpp]: Please, wrap the body by {}.

**Comment by ghost** [comment]
Reviewed

File: client/client-multi/client-cpp/impl/ClientStatus.cpp [hunk 2]
```diff
@@ -420,51 +420,59 @@ SequenceNumber ClientStatus::getAppSeqNumber() const
...
 
 std::string ClientStatus::getEndpointAccessToken()
 {
-    auto parameter_it = parameters_.find(ClientParameterT::EP_ACCESS_TOKEN);
-    if (parameter_it != parameters_.end()) {
-        auto token = boost::any_cast<std::string>(parameter_it->second->getValue());
-        if (!token.empty()) {
-            return token;
-        }
+    std::string token = "";
```
Comment (ohamov) [inline client/client-multi/client-cpp/impl/ClientStatus.cpp]: There is no need to initialize the variable explicitly. For this purposes std::string has default constructor.

**Comment by ohamov** [comment]
Minor nitpick, in general the code looks good to me.

**Pull request merged**